src/ape/api/providers.py

```diff
@@ -1316,6 +1386,109 @@ def _find_txn_by_account_and_nonce(
                     stop_block,
                 )
 
+    def poll_blocks(
+        self,
+        stop_block: Optional[int] = None,
+        required_confirmations: Optional[int] = None,
+        new_block_timeout: Optional[int] = None,
+    ) -> Iterator[BlockAPI]:
+        network_name = self.network.name
+        block_time = self.network.block_time
+        wait_time = block_time / 2
+        timeout = (
+            (
+                10.0
+                if network_name == LOCAL_NETWORK_NAME or network_name.endswith("-fork")
+                else 50 * block_time
+            )
+            if new_block_timeout is None
+            else new_block_timeout
+        )
+
+        if required_confirmations is None:
+            required_confirmations = self.network.required_confirmations
+
+        last_yielded_height = None
+
+        reorg_height = None
+
+        while True:
+            if (
+                stop_block is not None
+                and last_yielded_height is not None
+                and last_yielded_height >= stop_block
+            ):
+                break
+            latest_block = self.web3.eth.block_number
+            if last_yielded_height is None:
+                last_yielded_height = latest_block - required_confirmations
```

Comment:
**warning**: potentially dangerous to set a variable implying something was yield _before_ anything got yielded.. the actually yielding takes places much later. I can forsee an issue with this but I don't know if that is the problem or not

Reply:
* note, added comments and renames and things are little clearer now